### PR TITLE
Update free-courses-en.md (changes made)

### PR DESCRIPTION
**Pull request by HarshRanaOC**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add inf | Improve repo

### Why is this valuable (or not)?
I have taken this course.

### How do we know it's really free?
edX courses are for free and self paced.
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by HarshRanaOC** [comment]
Hello @sohansai, I've followed all the contribution guidelines and crafted a flawless pull request.

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -1232,6 +1233,7 @@
 * [Mathematics for Machine Learning - Linear Algebra](https://www.youtube.com/playlist?list=PLiiljHvN6z1_o1ztXTKWPrShrMrBLo5P3) - Imperial College London, Dr David Dye, Dr Sam Cooper
 * [Mathematics for Machine Learning - Multivariate Calclus](https://www.youtube.com/playlist?list=PLiiljHvN6z193BBzS0Ln8NnqQmzimTW23) - Imperial College London, Dr David Dye, Dr Sam Cooper
 * [Pattern Recognition and Machine Learning](https://www.microsoft.com/en-us/research/people/cmbishop/#!prml-book)
+* [Python Machine Learning & AI Mega Course ](https://www.youtube.com/watch?v=WFr2WgN9_xE&t=0s) - Tech With Tim
```
Comment (Thenlie) [inline courses/free-courses-en.md:1236]: For youtube, we prefer "playlist" links and the name of the producer/creator. View [these docs](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md#platforms-and-access-notes) for more information. Once you have updated you PR, you can add “changes made” to the title and reviewer will take another look as soon as they are available. Thanks!

**Review by Thenlie — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md (hunk 1 above)
Comment (HarshRanaOC) [reply courses/free-courses-en.md:1236]: Hello @Thenlie, the course consists of a single video that lasts for 7 hours. The creator hasn't organized it into a playlist.

**Review by Thenlie — APPROVED** [review]

**Comment by HarshRanaOC** [comment]
Hi @sohansai, I have made the required changes.
Regards

**Comment by HarshRanaOC** [comment]
Hi @sohansai . I have made the required changes, kindly mention if more have to be made.
Regards

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -1551,6 +1558,7 @@
 * [MOOC Introduction to Cyber Security 2023](https://cybersecuritybase.mooc.fi) - University of Helsinki
 * [Nightmare](https://guyinatuxedo.github.io/index.html) - guyinatuxedo
 * [Professor Messer’s SY0-601 CompTIA Security+ Course](https://www.professormesser.com/security-plus/sy0-601/sy0-601-video/sy0-601-comptia-security-plus-course/) - Professor Messer
+* [Python for Hackers FULL Course | Bug Bounty & Ethical Hacking](https://www.youtube.com/playlist?list=PLBfoYdk-WAlzfRFHRKAmA7Vw2svoGqfb5) - PhD Security
```
Comment (Thenlie) [inline courses/free-courses-en.md:1561]: I think this would fit better under the python heading.

Comment (HarshRanaOC) [reply courses/free-courses-en.md]: Hello @Thenlie, I wanted to let you know that I've carefully considered your suggestion and implemented the necessary adjustments. 
Regards

**Review by Thenlie — APPROVED** [review]

File: courses/free-courses-en.md (hunk 2 above)
Comment (Thenlie) [reply courses/free-courses-en.md]: Thanks for making these changes! 🚀

File: courses/free-courses-en.md [hunk 3]
```diff
@@ -1404,6 +1411,7 @@
 * [Python for Beginners (Full Course)](https://www.youtube.com/playlist?list=PLsyeobzWxl7poL9JTVyndKe62ieoN-MZ3) - Telusko
 * [Python for Data Science, AI & Development](https://www.coursera.org/learn/python-for-applied-data-science-ai) - Joseph Santarcangelo (Coursera)
 * [Python for Everybody](https://www.py4e.com/lessons) (Coursera Reading Materials with YouTube Videos)
+* [Python for Hackers FULL Course | Bug Bounty & Ethical Hacking](https://www.youtube.com/playlist?list=PLBfoYdk-WAlzfRFHRKAmA7Vw2svoGqfb5) - PhD Security
```
Comment (eshellman) [inline courses/free-courses-en.md:1414]:
```suggestion
* [Python for Hackers FULL Course \| Bug Bounty & Ethical Hacking](https://www.youtube.com/playlist?list=PLBfoYdk-WAlzfRFHRKAmA7Vw2svoGqfb5) - PhD Security
```

File: courses/free-courses-en.md [hunk 4]
```diff
@@ -307,6 +307,7 @@
 
 ### Blockchain
 
+* [BerkeleyX: Blockchain Technology](https://www.edx.org/learn/blockchain/university-of-california-berkeley-blockchain-technology?index=product&objectID=course-17b2db20-bb6d-4148-ad0b-9b9b99e287c1&webview=false&campaign=Blockchain+Technology&source=edX&product_category=course&placement_url=https%3A%2F%2Fwww.edx.org%2Flearn%2Fblockchain) - Rustie Lin and Nadir Akhtar
```
Comment (eshellman) [inline courses/free-courses-en.md:310]: do we know that the url doesn't change after it's archived? are all the url parameters neede? I get hte same page with https://www.edx.org/learn/blockchain/university-of-california-berkeley-blockchain-technology

File: courses/free-courses-en.md [hunk 5]
```diff
@@ -1234,6 +1236,7 @@
 * [Mathematics for Machine Learning - Linear Algebra](https://www.youtube.com/playlist?list=PLiiljHvN6z1_o1ztXTKWPrShrMrBLo5P3) - Imperial College London, Dr David Dye, Dr Sam Cooper
 * [Mathematics for Machine Learning - Multivariate Calclus](https://www.youtube.com/playlist?list=PLiiljHvN6z193BBzS0Ln8NnqQmzimTW23) - Imperial College London, Dr David Dye, Dr Sam Cooper
 * [Pattern Recognition and Machine Learning](https://www.microsoft.com/en-us/research/people/cmbishop/#!prml-book)
+* [Python Machine Learning & AI Mega Course ](https://www.youtube.com/playlist?list=PLBfoYdk-WAly5ib6M8ifOnDGl33bSvF_n) - Tech With Tim (Tim Ruscica)
```
Comment (eshellman) [inline courses/free-courses-en.md:1239]: Youtube attribution is "HARSHVARDHAN RANA"

**Review by eshellman — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md (hunk 5 above)
Comment (HarshRanaOC) [reply courses/free-courses-en.md]: Hello @eshellman, I wanted to let you know that the course consists of a single 7-hour long video without a playlist. So, I took the initiative to create one myself. However, I came across an even better course that already has a playlist curated by the creator. I've integrated it into the code.
Regards

Comment (eshellman) [reply courses/free-courses-en.md]: Aha. A 7-hour single video is fine, as long as you tell us that in the description of the PR (guideline is >1 hr.)

Comment (eshellman) [reply courses/free-courses-en.md]: you should switch it back to what you had before. sorry for the confusion.

Comment (HarshRanaOC) [reply courses/free-courses-en.md]: Hello @eshellman, I've re-written that course and also included the one by the same creator that I found to be more beneficial in the code.

**Pull request merged**